src/rail/estimation/algos/deepdisc.py

```diff
@@ -143,11 +145,41 @@ class DeepDiscInformer(CatInformer):
     def __init__(self, args, comm=None):
         CatInformer.__init__(self, args, comm=comm)
 
+    def inform(self, training_data, metadata):
+        self.set_data('input', training_data)
+        self.set_data('metadata', metadata)
+        self.run()
+        self.finalize()
+        return self.get_handle('model')
+
+    def finalize(self):
+        self.temp_dir.cleanup()
+
     def run(self):
         """
         Train a inception NN on a fraction of the training data
         """
         train_data = self.get_data("input")
+        metadata = self.get_data("metadata")
+
+        # create an iterator here
+        itr = self.input_iterator("input")
+        self.temp_dir = tempfile.TemporaryDirectory()
+        for start_idx, _, chunk in itr:
+            for idx, image in enumerate(chunk):
```

Comment:
Instead of `for idx, image in enumerate(chunk):`, let's pick a different name for the index variable. Perhaps `image_index`?

Reply:
going with `image_idx` for consistency with `start_idx`